projects/treasury/bitdao.js

```diff
@@ -1,112 +1,88 @@
-'0x8AA6a67e96850e070B0c8E94E3a35C5f9f01809C',
-'0x50f6e426fdefb3f994d3fe9fa4e1ee715f85de7f',
-'0x7427b4Fd78974Ba1C3B5d69e2F1B8ACF654fEB44',
-'0x7fe2bAffD481a8776A9eaD15a8eD17Fe37107903',
-'0x15Bb5D31048381c84a157526cEF9513531b8BE1e',
-'0xdD1c2483056fF46153249bd903401ae7bF6360D1',
-'0x565F603D583F9199487923775114ae8c0D17D044',
-'0x650aD9e7EfCD34B7d050c22a6A8dFFAFe3B4A22E',
-'0x607105cE5bf13e70B49E949a3DdFaD694d19374F',
-'0x131C7f3461A6696317ddfEdfed3BCdc10A2062B2',
-'0xa1F7D91Bf121f4940d96c5C52Bc577011B95B51b',
-'0x911169AA285f5D18fC3567d150616d4B0869d3a5',
-'0x3f946F00A00eB2A66A4BD1AeAF137E05dB6CAEc6',
-'0x9fe09b3ed1A407162876fEB1995048A620552fD0',
-'0xd4338fC8Dc9d2FDcb99604d3cFc80019EBE01058',
-'0x71Fb53Afc7E36C3f11BC1bdBBAB7B6FC3E552eb6',
-'0x92A9e359d72F934a5d7c1251201f9855A381B23c',
-'0xb118d4B94B0D4ce38F0D15d88f1dC09580a60b7A',
-'0xaA42736947d1fdcc5d93F459A6D1dC0d7b9a92a4',
-'0xF366eC9FA2DCE0ee7A6bdae4Aaa5c076E8391AFC',
-'0x5DA939F5e2bC3C7159ac16f98BbFb23759000cd5',
-'0x60F6ce1965D99EEffDF63B5303664f25fCb0347F',
-'0xC784F3aEA5ce3daBA6907ee5d6Ce321a204Eb3A8',
-'0xDCA65E2DFEe70991374eD47EfB4aD6B4FCD0c612',
-'0x4ea7b4D10a01c93509BaA0CBb128c89344A1F578',
-'0x4dF3d780Af7cbD51d9c76f236477edF417c7B554',
-'0xA38e519b12D9CE133396a3E4EB92aac0934AB351',
-'0x6d9755211D627fe0EA02D94C23C6110af16a8882',
-'0x43c0f24E84e6d45b021d18C982beAbFA969577c8',
-'0xB82C91bB7e8696a4A057192ED61aFcD1F9121722',
-'0x15FFBf5730FA9eF271B2E9b4a1a6c90F2288155B',
-'0xCef70f66e50CF016BB932De6425AA6f7286A3886',
-'0x50165383783124232B9e4367D59815947012Ac27',
-'0x97D50c7d14E68bEBC0f81B4FdCed89a1122330A6'
-]
+  const wallets = Object.values(
+    datas.reduce((acc, { id, walletAddress, amount }) => {
+      acc[walletAddress] = acc[walletAddress] || { owner: walletAddress, tokens: [] };
+      acc[walletAddress].tokens.push({ address: id, amount });
+      return acc;
+    }, {})
+  );
 
-const tokenTreasuries = treasuryExports({
+  const uniqueOwners = new Set();
+  const uniqueTokens = new Set();
+
+  for (const { owner, tokens } of wallets) {
+    const eigenLayerToken = tokens.find(token => token.address === 'eigen-layer-eth');
+    if (tokens.some(token => token.address === 'eth')) api.add(nullAddress, (await sdk.api.eth.getBalance({ target: owner })).output);
+    if (tokens.some(token => token.address === 'mnt')) api.add(MNT, (await sdk.api.eth.getBalance({ target: owner })).output, { skipChain: true });
+    if (tokens.some(token => token.address === 'ethena-farming-usde')) api.add(USDe, await getEthenaFarmingBalance(api, owner));
+    if (eigenLayerToken) api.add(nullAddress, eigenLayerToken.amount * 10 ** 18);
+
+    const nonSpecificTokens = tokens.filter(token => !isSpecificToken(token.address));
+    if (nonSpecificTokens.length > 0) {
+      uniqueOwners.add(owner);
+      nonSpecificTokens.forEach(token => uniqueTokens.add(token.address));
+    }
+  }
+
+  return { owners: Array.from(uniqueOwners), tokens: Array.from(uniqueTokens) };
```

Comment:
I sort of prefer, tokensAndOwners array in these cases because if it is not a common list of tokens, #calls grows exponentially, as it is #tokens x #owners